src/io/flutter/preview/PreviewView.java

```diff
@@ -322,6 +332,51 @@ private void selectPath(TreePath selectionPath, boolean focusEditor) {
         currentEditor.getCaretModel().addCaretListener(caretListener);
       }
     }
+
+    if (SHOW_PREVIEW_AREA) {
+      final Element buildMethodElement = getBuildMethodElement(selectionPath);
+      previewAreaPanel.updatePreviewElement(getElementParentFor(buildMethodElement), buildMethodElement);
+    }
+  }
+
+  // TODO: Add parent relationship info to FlutterOutline instead of this O(n^2) traversal.
+  private Element getElementParentFor(@Nullable Element element) {
+    if (element == null) {
+      return null;
+    }
+
+    for (FlutterOutline outline : outlineToNodeMap.keySet()) {
+      final List<FlutterOutline> children = outline.getChildren();
+      if (children != null) {
+        for (FlutterOutline child : children) {
+          if (child.getDartElement() == element) {
+            return outline.getDartElement();
+          }
+        }
+      }
+    }
+
+    return null;
+  }
+
+  private Element getBuildMethodElement(TreePath path) {
+    for (Object n : path.getPath()) {
+      final DefaultMutableTreeNode node = (DefaultMutableTreeNode)n;
+      final OutlineObject outlineElement = (OutlineObject)node.getUserObject();
+      if (outlineElement == null) {
+        continue;
+      }
+
+      final FlutterOutline flutterOutline = outlineElement.outline;
+
+      if (flutterOutline.getDartElement() != null) {
```

Comment:
We repeat `flutterOutline.getDartElement()` three times :)

Reply:
done